### PR TITLE
Changed registration to be a 2 step process.

### LINKED ISSUE #220: Password for restored accounts
Currently there is no way to add a password for an account that is created from a Seed phrase, this should be added to the new registration UI, this will depend on another issue with a proposal to overhaul the registration UI that will be created later so lets wait on this one

### LINKED ISSUE #232: Two Step registration UI
The current registration UI is quite cluttered, with the user having to take in lots of information. i think we should split the new account process into a two step ordeal.

1. The User should enter their desired username and password

image

2. The user should be presented with their seed 

image

#220 should be worked into this UI

### PR DESCRIPTION
**Pull request by Mikunj**
Resolves #232 
Resolves #220 

First screen:
Screen Shot 2019-04-04 at 11 09 32 am

Second screen:
Screen Shot 2019-04-04 at 11 09 36 am

### CODE REVIEW

File: background.html [hunk 1]
```diff
@@ -655,49 +655,63 @@ <h3>{{ message }}</h3>
...
-            <div class='standalone-register-buttons'>
-              <a class='button' id='generate-mnemonic'>Generate Seed</a>
-              <a class='button' id='copy-mnemonic'>Copy Seed</a>
-              <a class='button' id='register' data-loading-text='Please wait...'>Register</a>
+            <div class='buttons'>
+              <a class='button grey' id='back-button'>Back</a>
+              <a class='button' id='save-button' data-loading-text='Please wait...'>Save</a>
             </div>
           </div>
+
```
Comment (sachaaaaa) [inline background.html]: intentional?

Comment (Mikunj) [reply background.html]: Ye, i padded them a bit so you can clearly see the start and end, i was lost in divs 😭

File: js/views/standalone_registration_view.js [hunk 2]
```diff
@@ -50,14 +53,40 @@
...
       'change #mnemonic-display-language': 'onGenerateMnemonic',
       'click #copy-mnemonic': 'onCopyMnemonic',
       'click .section-toggle': 'toggleSection',
-      'keyup #password': 'onPasswordChange',
+      'keyup #password': 'onValidatePassword',
       'keyup #password-confirmation': 'onValidatePassword',
     },
+    async showPage(pageIndex) {
+      // eslint-disable-next-line func-names
+      this.$pages.each(function(index) {
```
Comment (BeaudanBrown) [inline js/views/standalone_registration_view.js]: Is there a reason this isn't using arrow function?

File: js/views/standalone_registration_view.js [hunk 3]
```diff
@@ -50,14 +53,40 @@
...
+        if (index !== pageIndex) {
+          $(this).hide();
+        } else {
+          $(this).show();
+        }
+      });
+    },
+    async showRegisterPage() {
+      this.registrationParams = {};
+      this.showPage(0);
```
Comment (BeaudanBrown) [inline js/views/standalone_registration_view.js]: Should we have an enum style object or something for these indexes?
(Argument against would be they are only used in `showXPage` functions anyway)

Comment (Mikunj) [reply js/views/standalone_registration_view.js]: Hmm, good point, i'm not sure if it's actually needed though unless we have more than 2 pages 🤔

File: js/views/standalone_registration_view.js (hunk 2 above)
Comment (Mikunj) [reply js/views/standalone_registration_view.js]: `$(this)` is required and using the arrow function will bind `this` to the registration view instead making it impossible to access the page instance

**Pull request merged**